### ISSUE TITLE
Doesn't work with real serial terminals

### ISSUE THREAD

**Issue by dhylands** [AndreRenaud/EmbeddedCLI#19]
When you press ENTER (or RETURN) on a serial terminal, it generates a CR and not NL.

I noticed that the posix demo works because it doesn't use raw mode, and leaves the CR->NL translation turned on.

If you use `screen` under linux over a serial port, it sends a CR when you press ENTER, and the embedded CLI doesn't recognize it as an end of line.

Similarly, on the output side, the code currently only generates a NL at the end of the line, and it really needs to generate a CR NL.

I can work around these things by translating CR to NL on the input side and translate NL to CR NL on the putchar side, but I thought I would report it as an issue, since if it were fixed then it would work properly with most of the terminal programs out there. Yes most terminal programs have the option to do this translation for you, but my thinking is that you shouldn't need to rely on the terminal program.

Perhaps it should be a configuration option?

I'm happy to code up a PR for your consideration.

**Comment by AndreRenaud** [comment]
Yes, that would be a good addition. This library hasn't seen a lot of work since I put it together originally, and the CR/LF issue is definitely a problem with serial ports. If you put together a PR, that would be great.

**Comment by dhylands** [comment]
I created #20

**Comment by AndreRenaud** [comment]
Thanks for that PR - I pushed a couple of commits on top of it to apply clang-format & fix one of the tests to cope with \r\n behaviour. Looks like I don't have the actions configured to run on pull-requests.

**Comment by dhylands** [comment]
Thanks for the code :)
I'm working for Kepler Communications and working on the attitude determination and control system for a new satellite. I'm integrating this into our code base.
